exercise/sitemaps.py

```diff
@@ -6,8 +8,21 @@
 from .models import BaseExercise, CourseChapter, LearningObject
 
 
+DELTA_DAYS_NEW = 180
```

Comment:
Could the duplicated code block be reused somehow? e.g. moving to a proper place and calling from there?

Reply:
Actually, should there be a common AplusSitemap base class for all these, because most different Sitemap implementations seem to follow similar pattern? Then it would be easier to maintain consistent behavior in different parts of A+ also in future.

Reply:
This would be good, I could move the function ``is_recent_or_upcoming`` and the constant value of 180 days into that class.

Would this class be in a new separate file? If so what directory do you think the new file should be placed in?

Reply:
I think lib directory is a typical place for common things.

There are also other commonalities: location function looks the same in most classes, so it could be defined in base class. Also the priority function follows same pattern everywhere:
return 2*base_priority if is_new(item) else base_priority.
So if the subclasses just define the base priority, then the same priority function implementation would work if implemented in base class.

Reply:
I saw one case (CourseStaticViewSitemap) with static priority. Maybe there the priority function could be overridden to return a static value. If it was designed this way, the subclasses should anyway define some variable ("base_priority" or "weight", or something like that) to be used in calculation in the priority function.